packages/itwin/ec3-widget/src/components/EC3Router.tsx

```diff
@@ -0,0 +1,47 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import React from "react";
+import type { Route } from "./EC3Widget";
+import { RouteStep } from "./EC3Widget";
+import { TemplateMenu } from "./TemplateMenu";
+import { Templates } from "./Templates";
+
+export const EC3Router = ({
+  currentRoute,
+  navigateTo,
+  goBack,
+}: {
+  currentRoute: Route;
+  navigateTo: (toRoute: (prev: Route | undefined) => Route) => void;
+  goBack: () => void;
+}) => {
+  const { template } = currentRoute.ec3RoutingFields;
+  switch (currentRoute.step) {
+    case RouteStep.Templates:
+      return (
+        <Templates
+          onClickCreate={() =>
+            navigateTo(() => ({
+              step: RouteStep.TemplateMenu,
+              title: "Create Template",
+              ec3RoutingFields: {},
```

Comment:
Why does this need to be named `ec3RoutingFields`? Can the name be simplified to `routingFields` or `routingArguments`?

Reply:
Changed to `routingFields`